pkg/client/cache/delta_fifo.go

```diff
@@ -334,26 +336,42 @@ func (f *DeltaFIFO) Replace(list []interface{}) error {
 				continue
 			}
 		}
-		if err := f.queueActionLocked(Deleted, DeletedFinalStateUnknown{k}); err != nil {
+		v, exists, err := f.knownObjects.GetByKey(k)
+		if err != nil || !exists {
+			v = nil
+			glog.Infof("Unable to lookup key %v returned by list, ignoring", k)
```

Comment:
We need to preserve the signature of GetByKey for the store interface, I've added the error log line